File: Medium/55. Jump Game/JumpGame.java
/*
LeetCode Medium: 55. Jump Game
https://leetcode.com/problems/jump-game/

Check JumpGame.cpp for brief solution explanation.
*/
class Solution {
    public boolean canJump(int[] nums) {
        int N = nums.length, mx_reachable = 0;
        for(int i = 0;i<N;i++){
            if(mx_reachable<i){
                return false;
            }else{
                mx_reachable = Math.max(mx_reachable, i+nums[i]);
            }
        }
        return true;
    }
}

File: Medium/55. Jump Game/JumpGame.py
'''
LeetCode Medium: 55. Jump Game
https://leetcode.com/problems/jump-game/

Check JumpGame.cpp for brief solution explanation.
'''
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        N = len(nums)
        mx_reachable = 0
        for i in range(N):
            if mx_reachable<i:
                return False
            else:
                mx_reachable = max(mx_reachable, i+nums[i])
        return True